helix-core/src/match_brackets.rs

```diff
@@ -28,6 +28,11 @@ pub fn find(syntax: &Syntax, doc: &Rope, pos: usize) -> Option<usize> {
     let end_byte = node.end_byte() - 1; // it's end exclusive
     let start_char = doc.byte_to_char(start_byte);
     let end_char = doc.byte_to_char(end_byte);
+    let len = doc.len_chars();
+
+    if start_char >= len || end_char >= len {
```

Comment:
I think `start_char` and `end_char` should always be valid, we got them straight from `byte_to_char`. If they were out of bounds, then `byte_to_char` would panic.

Reply:
I cannot reproduce #107 but the command 16xd should panic.